tests/rptest/services/compacted_verifier.py

```diff
@@ -0,0 +1,222 @@
+
+
+class Workload(Enum):
+    IDEMPOTENCY = 1
+
+
+workloads = {Workload.IDEMPOTENCY: "io.vectorized.compaction.idempotency.App"}
+
+
+class CompactedVerifier(Service):
+    logs = {"rw_stdout_stderr": {"path": OUTPUT_LOG, "collect_default": True}}
+
+    def __init__(self, context, redpanda, workload):
+        super(CompactedVerifier, self).__init__(context, num_nodes=1)
+        self._redpanda = redpanda
+        self._is_done = False
+        self._node = None
+        if workload not in Workload:
+            raise Exception(f"Unknown workload {workload}")
+        self._workload = workload
+
+    def is_alive(self, node):
+        result = node.account.ssh_output(
+            "bash /opt/remote/control/alive.sh rw")
+        result = result.decode("utf-8")
+        return "YES" in result
+
+    def is_ready(self):
+        try:
+            self.remote_ping()
+            return True
+        except requests.exceptions.ConnectionError:
+            return False
+
+    def ensure_progress(self, delta, timeout_sec):
+        if not self.is_alive(self._node):
+            self.raise_on_violation(self._node)
+            raise CrushedException(
+                "rw_verifier is crushed (it often happens on the consistency violation)"
+            )
+
+        old_state = self.remote_info()
+        min_writes = old_state["min_writes"]
+
+        def check_writes():
+            new_state = self.remote_info()
+            if new_state["min_writes"] - min_writes < delta:
+                return False
+            return True
+
+        wait_until(
+            check_writes,
+            timeout_sec,
+            2,
+            err_msg=
+            f"writes got stuck: hasn't written {delta} records in {timeout_sec}s"
+        )
+
+    def has_cleared(self, threshold, timeout_sec):
+        if not self.is_alive(self._node):
+            self.raise_on_violation(self._node)
+            raise CrushedException(
+                "rw_verifier is crushed (it often happens on the consistency violation)"
+            )
+
+        def check():
+            new_state = self.remote_info()
+            if new_state["min_writes"] < threshold:
+                return False
+            return True
+
+        wait_until(
+            check,
+            timeout_sec,
+            2,
+            err_msg=
+            f"num of rw iteration hasn't reached {threshold} in {timeout_sec}")
+
+    def raise_on_violation(self, node):
+        self.logger.info(
+            f"Scanning node {node.account.hostname} log for violations...")
+
+        for line in node.account.ssh_capture(
+                f"grep -e violation {OUTPUT_LOG} || true"):
+            raise ConsistencyViolationException(line)
+
+    ### Service overrides
+
+    def start_node(self, node, timeout_sec=10):
+        node.account.ssh(
+            f"bash /opt/remote/control/start.sh rw \"java -cp /opt/verifiers/verifiers.jar {workloads[self._workload]}\""
+        )
+        sleep(1)
```

Comment:
do we need this ?

Reply:
oh, I don't remember why, I reused the code I wrote for https://github.com/redpanda-data/redpanda/pull/5238 and I'm pretty sure I added it without a reason

Reply:
maybe it would be better to use `wait_until` here ? Relaying on sleep with a particular duration is usually a source of flakiness

Reply:
the next line is `wait_until` so it shouldn't matter but ok let me drop it, rerun the test and see if there was a reason to add it

Reply:
it looks like it isn't needed, dropping